### PR TITLE
add late fusion fpn backbone for 2.5D models

### PR DESCRIPTION
**Pull request by nanddalal**
Thanks for your contribution!

If you're sending a large PR (e.g., >50 lines),
please open an issue first about the feature / bug, and indicate how you want to contribute.
See more at https://detectron2.readthedocs.io/notes/contributing.html#pull-requests
about how we handle PRs.

Before submitting a PR, please run `dev/linter.sh` to lint the code.

### CODE REVIEW

File: detectron2/modeling/proposal_generator/rrpn_outputs.py [hunk 1]
```diff
@@ -103,7 +107,9 @@ def find_top_rrpn_proposals(
 
         topk_proposals.append(topk_proposals_i)
         topk_scores.append(topk_scores_i)
-        level_ids.append(torch.full((num_proposals_i,), level_id, dtype=torch.int64, device=device))
+        level_ids.append(
+            torch.full((num_proposals_i,), level_id, dtype=torch.int64, device=device)
+        )
```
Comment (nanddalal) [inline detectron2/modeling/proposal_generator/rrpn_outputs.py]:
```suggestion
        level_ids.append(torch.full((num_proposals_i,), level_id, dtype=torch.int64, device=device))
```

File: detectron2/modeling/proposal_generator/rpn_outputs.py [hunk 2]
```diff
@@ -115,7 +119,9 @@ def find_top_rpn_proposals(
 
         topk_proposals.append(topk_proposals_i)
         topk_scores.append(topk_scores_i)
-        level_ids.append(torch.full((num_proposals_i,), level_id, dtype=torch.int64, device=device))
+        level_ids.append(
+            torch.full((num_proposals_i,), level_id, dtype=torch.int64, device=device)
+        )
```
Comment (nanddalal) [inline detectron2/modeling/proposal_generator/rpn_outputs.py]:
```suggestion
        level_ids.append(torch.full((num_proposals_i,), level_id, dtype=torch.int64, device=device))
```

File: detectron2/modeling/meta_arch/rcnn.py [hunk 3]
```diff
@@ -173,14 +181,63 @@ def inference(self, batched_inputs, detected_instances=None, do_postprocess=True
             results = self.roi_heads.forward_with_given_boxes(features, detected_instances)
 
         if do_postprocess:
-            return GeneralizedRCNN._postprocess(results, batched_inputs, images.image_sizes)
+            if self.late_fusion:
+                images_sizes = images[0].image_sizes
```
Comment (nanddalal) [inline detectron2/modeling/meta_arch/rcnn.py]:
```suggestion
                image_sizes = images[0].image_sizes
```

File: detectron2/modeling/meta_arch/rcnn.py [hunk 4]
```diff
@@ -173,14 +181,63 @@ def inference(self, batched_inputs, detected_instances=None, do_postprocess=True
...
+            else:
+                images_sizes = images.image_sizes
```
Comment (nanddalal) [inline detectron2/modeling/meta_arch/rcnn.py]:
```suggestion
                image_sizes = images.image_sizes
```

File: detectron2/modeling/backbone/fpn.py [hunk 5]
```diff
@@ -210,14 +280,25 @@ def build_resnet_fpn_backbone(cfg, input_shape: ShapeSpec):
...
     out_channels = cfg.MODEL.FPN.OUT_CHANNELS
-    backbone = FPN(
-        bottom_up=bottom_up,
-        in_features=in_features,
-        out_channels=out_channels,
-        norm=cfg.MODEL.FPN.NORM,
-        top_block=LastLevelMaxPool(),
-        fuse_type=cfg.MODEL.FPN.FUSE_TYPE,
-    )
+    if cfg.MODEL.LATE_FUSION.ENABLED:
```
Comment (kaizhang-nines) [inline detectron2/modeling/backbone/fpn.py:283]: MODEL.LATE_FUSION is optional. I don't see that you modified the default config, so it's better to have a default value "False" using `get`.

**Review by kaizhang-nines — CHANGES_REQUESTED** [review]
Please either add MODEL.LATE_FUSION to the default config, or give it a False default value.

**Review by kaizhang-nines — APPROVED** [review]
LGTM

**Pull request merged**